build-logic/src/main/kotlin/Android.kt

```diff
@@ -21,6 +23,10 @@ fun Project.configureAndroid(
         getCatalogVersion("android.sdkversion.min").toInt()
       }
       testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+
+      if (this is LibraryVariantDimension) {
+        multiDexEnabled = true
```

Comment:
Why do we need this?

Reply:
The test APK grew over 64k methods 🤷 . 

<img width="1267" alt="Screenshot 2024-03-25 at 13 30 53" src="https://github.com/apollographql/apollo-kotlin/assets/3974977/714bc788-8c44-4a7c-88b4-b9367b67a299">
<img width="1267" alt="Screenshot 2024-03-25 at 13 31 06" src="https://github.com/apollographql/apollo-kotlin/assets/3974977/c1746342-1478-4604-aaf6-0477143de417">

Interestingly, it's not the case on those 2 screenshots (I think? but somehow two dex files are still required 🤷 )
In all cases, it's close enough that it may happen again. Plus AGP enables it by default if `minSdk > 21` anyways. 

No place for degrowth in dex-land 🌱 🙈